### DISCUSSION TITLE
Strange problem with bash_completion@2

### DISCUSSION THREAD

**Discussion by piotr-sikora-v** [Homebrew/discussions#5490] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c98a33a376e5b35ebc801d0f7d6ba7b9eb118ac
Last commit: 6 days ago
Core tap JSON: 14 Jul 02:13 UTC
Core cask tap JSON: 14 Jul 02:13 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
brew doctor
Your system is ready to brew.
```

### Description of issue

When I start new console completion (i.e. in ssh) don't work.
To work I must in bash run another bash, then one more time exec:  `. "/opt/homebrew/etc/profile.d/bash_completion.sh"`
And after that completion works.

in `.bash_profile` i have:

```
eval "$(/opt/homebrew/bin/brew shellenv)"

export TERM=xterm
export PATH="/opt/homebrew/bin:${PATH}"

export BASH_COMPLETION_COMPAT_DIR="/opt/homebrew/etc/bash_completion.d/"
[[ -r "/opt/homebrew/etc/profile.d/bash_completion.sh" ]] && . "/opt/homebrew/etc/profile.d/bash_completion.sh"


export CLICOLOR=1

export PS1='\[\e[31m\]\u@\h:\w\[\e[0m\] '

cd ~/
. "/opt/homebrew/etc/profile.d/bash_completion.sh"
```

**Comment by gromgit** [comment]
> ```
> export BASH_COMPLETION_COMPAT_DIR="/opt/homebrew/etc/bash_completion.d/"
> ```

Is this actually needed?

> ```
> [[ -r "/opt/homebrew/etc/profile.d/bash_completion.sh" ]] && . "/opt/homebrew/etc/profile.d/bash_completion.sh"
> ```

Try adding this to your `~/.bashrc` as well. If I'm not mistaken, the vast majority of terminal software do _not_ spawn login shells when opening new session windows, hence `~/.bash_profile` will never be sourced.
